File: frontend/src/api/services.js
import client from './client';

export const servicesApi = {
  getServices: async (params = {}) => {
    const response = await client.get('/services', { params });
    return response.data;
  },

  getService: async (serviceId) => {
    const response = await client.get(`/services/${serviceId}`);
    return response.data;
  },

  createService: async (serviceData) => {
    const response = await client.post('/services', serviceData);
    return response.data;
  },

  updateService: async (serviceId, serviceData) => {
    const response = await client.put(`/services/${serviceId}`, serviceData);
    return response.data;
  },

  deleteService: async (serviceId) => {
    const response = await client.delete(`/services/${serviceId}`);
    return response.data;
  }
};

File: backend/routes/services.py
from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session
from typing import List
import models, schemas
from database import get_db

router = APIRouter()

@router.post("", response_model=schemas.Service)
def create_service(service: schemas.ServiceCreate, db: Session = Depends(get_db)):
    db_service = models.Service(**service.model_dump())
    db.add(db_service)
    db.commit()
    db.refresh(db_service)
    return db_service

@router.get("", response_model=List[schemas.Service])
def read_services(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):
    services = db.query(models.Service).filter(models.Service.is_active == True).offset(skip).limit(limit).all()
    return services

@router.get("/{service_id}", response_model=schemas.Service)
def read_service(service_id: int, db: Session = Depends(get_db)):
    service = db.query(models.Service).filter(models.Service.id == service_id).first()
    if service is None:
        raise HTTPException(status_code=404, detail="Service not found")
    return service

@router.put("/{service_id}", response_model=schemas.Service)
def update_service(service_id: int, service_update: schemas.ServiceCreate, db: Session = Depends(get_db)):
    db_service = db.query(models.Service).filter(models.Service.id == service_id).first()
    if db_service is None:
        raise HTTPException(status_code=404, detail="Service not found")
    
    for var, value in vars(service_update).items():
        setattr(db_service, var, value) if value else None
        
    db.commit()
    db.refresh(db_service)
    return db_service

@router.delete("/{service_id}")
def delete_service(service_id: int, db: Session = Depends(get_db)):
    db_service = db.query(models.Service).filter(models.Service.id == service_id).first()
    if db_service is None:
        raise HTTPException(status_code=404, detail="Service not found")
    db_service.is_active = False # Soft delete
    db.commit()
    return {"message": "Service deactivated"}
